detektive/src/main/kotlin/cash/detektive/javacompat/AnnotatePublicApisWithJvmOverloads.kt

```diff
@@ -50,14 +56,24 @@ class AnnotatePublicApisWithJvmOverloads(config: Config) : Rule(config) {
     }
     if (isApplicable(element, bindingContext[BindingContext.DECLARATION_TO_DESCRIPTOR, element])) {
       if (!element.annotationEntries.any { it.isOfType(JvmOverloads::class) }) {
-        report(
-          CodeSmell(
-            issue,
-            Entity.atName(element),
-            "Public ${elementType.name.lowercase()} '${element.nameAsSafeName}' " +
-              "with default arguments, but without @JvmOverloads annotation"
-          )
-        )
+        val message = "Public ${elementType.name.lowercase()} '${element.nameAsSafeName}' " +
+          "with default arguments, but without @JvmOverloads annotation"
+
+        if (autoCorrect) {
+          if (!(element as KtElement).isSuppressedBy(ruleId, aliases, ruleSetConfig.parentPath)) {
+            val annotation = element.addAnnotationEntry(KtPsiFactory.contextual(element.parent, markGenerated = true).createAnnotationEntry("@JvmOverloads"))
+            if (elementType == ElementType.CONSTRUCTOR) {
+              annotation.addBefore(KtPsiFactory.contextual(element.parent, markGenerated = true).createWhiteSpace(), null)
+              element.addAfter(KtPsiFactory.contextual(element.parent, markGenerated = true).createWhiteSpace(), null)
+            }
+            else if (elementType == ElementType.FUNCTION) {
+              annotation.addBefore(KtPsiFactory.contextual(element.parent, markGenerated = true).createNewLine(), null)
+            }
```

Comment:
Adding whitespace - without these, for some reason, the annotation on constructors ends up being added as `<ClassName>@JvmOverloadsconstructor` (all in one word) which doesn't compile. Figuring out how to add whitespace was a bit of trail and error and it messes up with formatting a bit for functions.